addons/account_payment/models/payment_transaction.py

```diff
@@ -186,8 +198,10 @@ def _log_message_on_linked_documents(self, message):
         """
         self.ensure_one()
         self = self.with_user(SUPERUSER_ID)  # Log messages as 'OdooBot'
-        if self.source_transaction_id.payment_id:
-            self.source_transaction_id.payment_id.message_post(body=message)
+        if self.source_transaction_id:
+            payment_id = self.source_transaction_id.payment_id
+            if payment_id:
+                payment_id.message_post(body=message)
```

Comment:
Hm, no, because we still need to post the message to the `source_tx`'s invoices if there isn't a `payment_id`.

I put the `for` loop before the `if` to make it clearer =]

We didn't put that with the refund feature because before, we linked the invoices to the refund tx, which isn't the case anymore.